### PR TITLE
Capz should not add tags to managed resource group

### PR DESCRIPTION
**Pull request by LochanRn**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

When delete capz cluster is triggered, capz try to delete the resource group, even through the RG is not created by capz. 
Capz should check if err and only continue to createOrUpdate RG when err is resource-not-found. otherwise when azure API have issue thus return some other error, the RG will be updated with owner-tags, then capz will think the RG is managed. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes None

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:
- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:
```release-note
None
```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/ok-to-test

**Comment by CecileRobertMichon** [comment]
Thanks for the PR @LochanRn. Can you please fix the merge conflict and add a unit test case that verifies the scenario you are fixing?

File: cloud/services/groups/groups.go [hunk 1]
```diff
@@ -55,10 +55,16 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	ctx, span := tele.Tracer().Start(ctx, "groups.Service.Reconcile")
 	defer span.End()
 
-	if _, err := s.client.Get(ctx, s.Scope.ResourceGroup()); err == nil {
+	_, err := s.client.Get(ctx, s.Scope.ResourceGroup())
+	if err == nil {
```
Comment (CecileRobertMichon) [inline cloud/services/groups/groups.go:59]: isn't this equivalent to what we were doing before with `if _, err := s.client.Get(ctx, s.Scope.ResourceGroup()); err == nil {` ?

Comment (LochanRn) [reply cloud/services/groups/groups.go:59]: Yes, but err will be local to `if _, err := s.client.Get(ctx, s.Scope.ResourceGroup()); err == nil {` block.
We will need the err in `if !azure.ResourceNotFound(err) {
 		return err
 	}
`
I will modify the PR with an else if block, to minimise the number of lines changed.

**Comment by LochanRn** [comment]
closing this ticket as I raised a fresh PR #1218 without merge conflict and added the required changes suggested based on the reviews

**Pull request closed without merging**